sd/etcd/example_test.go

```diff
@@ -0,0 +1,99 @@
+package etcd
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"golang.org/x/net/context"
+)
+
+// Package sd/etcd provides a wrapper around the coroes/etcd key value store (https://github.com/coreos/etcd) 
+// This example assumes the user has an instance of etcd installed and running locally on port 2379
+func Example() {
+
+	var logger log.Logger
+	{
+		logger = log.NewLogfmtLogger(os.Stdout)
+		logger = log.NewContext(logger).With("ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	}
+
+	// Creating random number too append to assure differentiation of test data keys 
+	rand.Seed(int64(time.Now().Nanosecond()))
+
+	value := rand.Int()
+	testData := fmt.Sprintf("foo%d", value)
+
+	value2 := rand.Int()
+	testData2 := fmt.Sprintf("foo%d", value2)
+
+
+	// Using basic options for NewClient
+	kitClientOptions := ClientOptions{
+		Cert:                    "",
+		Key:                     "",
+		CaCert:                  "",
+		DialTimeout:             (2 * time.Second),
+		DialKeepAline:           (2 * time.Second),
+		HeaderTimeoutPerRequest: (2 * time.Second)}
+
+	// Instantiate NewClient on localhost port 2379. Change this to whichever port you host etcd from 
+	kitClient, err := NewClient(context.Background(), []string{"http://:2379"}, kitClientOptions)
```

Comment:
There is no regular `clientOptions` or `client`, so we can use shorter names and construction forms here. Also, if Cert/Key/CaCert are optional, better to elide them altogether than set them to empty strings. Also, `DialKeepAline` typo?

```go
client, err := NewClient(context.Background(), []string{"http://:2379"}, ClientOptions{
    DialTimeout:             2 * time.Second,
    DialKeepAlive:           2 * time.Second,
    HeaderTimeoutPerRequest: 2 * time.Second,
})
```